akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -533,6 +533,28 @@ object Dropped {
     Dropped(message, reason, ActorRef.noSender, recipient)
 }
 
+object WrappedMessage {
+
+  /**
+   * Unwrap [[WrappedMessage]] recursively.
+   */
+  @tailrec def unwrap(message: Any): Any = {
+    message match {
+      case w: WrappedMessage => unwrap(w.message)
+      case _                 => message
+
+    }
+  }
+}
+
+/**
+ * Message envelopes may implement this trait for better logging, such as logging of
+ * message class name of the wrapped message instead of the envelope class name.
+ */
+trait WrappedMessage {
+  def message: Any
+}
```

Comment:
Would it be better to extract this trait to a dedicated file?

Reply:
we already have similar/related here, such as DeadLetter and Dropped

Reply:
akka.cluster.pubsub.WrappedMessage